public/app/features/alerting/unified/api/alertingApi.ts

```diff
@@ -0,0 +1,47 @@
+import { BaseQueryFn, createApi } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, getBackendSrv } from '@grafana/runtime';
+
+import { PromBuildInfoResponse } from '../../../../types/unified-alerting-dto';
+
+import { discoverAlertmanagerFeatures } from './buildInfo';
+
+const backendSrvBaseQuery = (): BaseQueryFn<BackendSrvRequest> => async (requestOptions) => {
+  try {
+    const response = await lastValueFrom(getBackendSrv().fetch<PromBuildInfoResponse>(requestOptions));
+
+    return { data: response.data };
+  } catch (error) {
+    return {
+      error: {
+        data: error,
+      },
+    };
+  }
+};
+
+export const alertingApi = createApi({
+  reducerPath: 'alertingApi',
+  baseQuery: backendSrvBaseQuery(),
+  endpoints: (build) => ({
+    discoverAmFeatures: build.query({
+      queryFn: async ({ amSourceName }: { amSourceName: string }) => {
+        try {
+          const amFeatures = await discoverAlertmanagerFeatures(amSourceName);
+          return { data: amFeatures };
+        } catch (error) {
+          return { error: error };
+        }
+      },
+    }),
+  }),
+});
+
+const { useDiscoverAmFeaturesQuery } = alertingApi;
```

Comment:
I think there's a lot of boilerplate overhead here with re-exporting properties – should we perhaps try to just work with the exported `alertingApi`? I feel like it's a bit too early to introduce additional abstractions

Reply:
Yes, I think we can use the injectEndpoints method to provide endpoints separation without much toil